### PR TITLE
Add soname to runtime library

### PR DESCRIPTION
**Pull request by bkmgit**
Fixes https://github.com/lfortran/lfortran/issues/2096

### CODE REVIEW

**Comment by bkmgit** [comment]
@certik @Pranavchiku Let me know if anything else is needed.

**Comment by certik** [comment]
In this PR, now the version inside LFortran does not work anymore:
```
$ lfortran --version
LFortran version: 0.1-git
Platform: macOS ARM
Default target: arm64-apple-darwin21.3.0
```

File: CMakeLists.txt [hunk 1]
```diff
@@ -11,9 +11,14 @@ if (LFORTRAN_BUILD_ALL)
     endif ()
 endif()
 
-file(STRINGS "${CMAKE_CURRENT_SOURCE_DIR}/version" LFORTRAN_VERSION)
+file(STRINGS "${CMAKE_CURRENT_SOURCE_DIR}/version" LFORTRAN_FULL_VERSION)
```
Comment (certik) [inline CMakeLists.txt:14]: To keep things simple, let's just use the name `LFORTRAN_VERSION`, that way the change will be minimal and the version inside the C++ code will start working again.

If a rename of this variable is needed, let's do it in a separate PR, after this is merged.

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -191,20 +191,23 @@ jobs:
         run: |
             cd scipy/
             micromamba activate scipy-dev
+            cp ${{ github.workspace }}/src/runtime/liblfortran_runtime.*  $CONDA_PREFIX/lib
```
Comment (certik) [inline .github/workflows/CI.yml:194]: Can these be moved above into the previous section, so that it looks like:
```
            LIBRARY_PATH="`pwd`/../../src/runtime/"
            FC=$(pwd)/../../src/bin/lfortran cmake \
              -DCMAKE_Fortran_FLAGS=--verbose \
              -DLFORTRAN_RUNTIME_LIBRARY_PATH=$LIBRARY_PATH \
              ..
            make install
            cp ${{ github.workspace }}/src/runtime/liblfortran_runtime.*  $CONDA_PREFIX/lib
```
This should work, right? Then we can remove it from the subsequent sections.

**Comment by certik** [comment]
Thanks for this, last couple issues, see above.

**Comment by bkmgit** [comment]
@certik Changes made. Rebased on main.

File: CMakeLists.txt (hunk 1 above)
Comment (bkmgit) [reply CMakeLists.txt]: Used `LFORTRAN_VERSION`

**Comment by certik** [comment]
The PR now looks very simple. The `lfortran` version seems to work now:
```
$ lfortran --version
LFortran version: 0.20.3-400-g308ca3160
Platform: macOS ARM
Default target: arm64-apple-darwin21.3.0
```
Let me continue testing this by hand a little more to see if I discover any other issues.

**Review by certik — APPROVED** [review]
I think this works. I tested a few things locally, it seems to work. Thanks @bkmgit for figuring this out!

Which linux distributions do you plan to put this into? Make sure to compile in Release mode with all optimizations on, for fast compilation.

**Comment by bkmgit** [comment]
Thanks. Fedora.

**Comment by certik** [comment]
@bkmgit perfect, once you submit it to Fedora, you can ping me and I can help review the build script.

I just made a release: https://github.com/lfortran/lfortran/releases/tag/v0.21.0 that contains your PR, so you can use that.

**Comment by certik** [comment]
Ha, it discovered a problem:
```
$ ./build1.sh 
++ pwd
+ cmake -DCMAKE_BUILD_TYPE=Debug -DWITH_LLVM=yes -DLFORTRAN_BUILD_ALL=yes -DWITH_STACKTRACE=yes -DWITH_RUNTIME_STACKTRACE=yes '-DCMAKE_PREFIX_PATH=;/Users/ondrej/mambaforge/envs/lf' -DCMAKE_INSTALL_PREFIX=/Users/ondrej/repos/lfortran/lfortran/inst .
+ ci/version.sh
++ git describe --tags --dirty
+ version=v0.21.0
+ version=0.21.0
+ echo 0.21.0
+ python src/libasr/asdl_cpp.py grammar/AST.asdl src/lfortran/ast.h
+ python src/libasr/asdl_cpp.py src/libasr/ASR.asdl src/libasr/asr.h
+ python src/libasr/wasm_instructions_visitor.py
Assuming default values of wasm_instructions.txt and wasm_visitor.h
+ cd src/lfortran/parser
+ re2c -W -b tokenizer.re -o tokenizer.cpp
+ cd src/lfortran/parser
+ re2c -W -b preprocessor.re -o preprocessor.cpp
+ cd src/lfortran/parser
+ bison -Wall -d -r all parser.yy
+ grep -n ''\''' src/lfortran/parser/parser.yy
OK
+ echo OK
CMake Error at CMakeLists.txt:17 (project):
  VERSION "0.21.0CACHESTRINGproject version" format invalid.


-- Configuring incomplete, errors occurred!
```
@bkmgit can you please submit a PR to fix it? We need to iterate on this. I'll make a new release once you submit.

**Comment by bkmgit** [comment]
Yes will make a fix. Sorry.
